### PR TITLE
Jhan/mobile annotation component

### PR DESCRIPTION
**Pull request by jhan6501**
# Description
Created a frontend mobile annotation component that features drawing on the screen for mobile optimization
Added a switch into stateManager to allow developers to quickly choose between using mobileAnnotation or webAnnotation (mobileannotation frontend works, but it's not linked with backend. webAnnotation works 100%, but the UI/UX isn't perfect yet)

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

## Before and After
https://user-images.githubusercontent.com/40724569/127545167-c6b9b9e6-166c-478d-b87e-5fd80a717920.mp4

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Checklist:

- [ ] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: droidlet/dashboard/web/src/StateManager.js [hunk 1]
```diff
@@ -131,6 +131,7 @@ class StateManager {
       objects: false,
       pose: false,
     };
+    this.useDesktopComponentOnMobile = false; // switch to use either desktop or mobile annotation on mobile device
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/StateManager.js:134]: Can you add a TODO or NOTE that the mobile annotation UX on mobile is incomplete?

File: droidlet/dashboard/web/src/components/MobileAnnotationComponents/DataEntry.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  submit() {
+    let name = this.nameRef.current.value.trim();
+    // Validation
+    if (name.length === 0) {
+      alert("You must enter the object's name");
+      return;
+    } else if (this.tags.length < 2) {
+      alert(
+        "Please enter at least 2 descriptive tags for the object. Examples include: color, shape, size, orientation, etc."
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/components/MobileAnnotationComponents/DataEntry.jsx:76]: Do instructions anywhere specify this has to be comma separated?

File: droidlet/dashboard/web/src/components/MobileAnnotationComponents/DataEntry.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      );
+      return;
+    }
+    this.props.onSubmit({
+      name: name.toLowerCase(),
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/components/MobileAnnotationComponents/DataEntry.jsx:81]: Can be useful to preserve capitals, eg. for proper nouns

File: droidlet/dashboard/web/src/components/MobileAnnotationComponents/ImageDrawer.jsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+import DataEntry from "./DataEntry";
+
+class ImageDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.canvasRef = React.createRef();
+    this.imgRef = React.createRef();
+    this.isDrawing = false;
+    this.annotationName = null;
+    this.annotationTags = null;
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/components/MobileAnnotationComponents/ImageDrawer.jsx:12]: why are some of these null and others empty list?

File: droidlet/dashboard/web/src/components/MobileAnnotationComponents/ImageDrawer.jsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+          this.ctx.closePath();
+          this.ctx.stroke();
+        }
+      }
+    }
+  };
+
+  // submit annotation to backend
+  dataEntrySubmit(objectData) {
+    this.drawing_data = objectData;
```
Comment (Rebecca-Qian) [inline droidlet/dashboard/web/src/components/MobileAnnotationComponents/ImageDrawer.jsx:99]: Nit: why is this snake case

**Review by Rebecca-Qian — APPROVED** [review]

File: droidlet/dashboard/web/src/components/MobileAnnotationComponents/DataEntry.jsx (hunk 3 above)
Comment (jhan6501) [reply droidlet/dashboard/web/src/components/MobileAnnotationComponents/DataEntry.jsx:81]: This file was taken directly from the existing component in AnnotationComponents, so for parallelism I think it's better to keep them the same

File: droidlet/dashboard/web/src/components/MobileAnnotationComponents/ImageDrawer.jsx (hunk 4 above)
Comment (jhan6501) [reply droidlet/dashboard/web/src/components/MobileAnnotationComponents/ImageDrawer.jsx:12]: Name is a string, tag is an array. will change

**Pull request merged**